xla/backends/profiler/gpu/rocm_tracer.cc

```diff
@@ -508,6 +512,24 @@ void RocmApiCallbackImpl::AddKernelEventUponApiExit(uint32_t cbid,
       event.kernel_info.func_ptr = (void*)func_addr;
       event.device_id = hipGetStreamDeviceId(stream);
     } break;
+    case HIP_API_ID_hipExtLaunchKernel: {
+      const void* func_addr = data->args.hipExtLaunchKernel.function_address;
+      hipStream_t stream = data->args.hipExtLaunchKernel.stream;
+      if (func_addr != nullptr)
+        event.name = hipKernelNameRefByPtr(func_addr, stream);
+
+      event.kernel_info.dynamic_shared_memory_usage =
+          data->args.hipExtLaunchKernel.sharedMemBytes;
+      event.kernel_info.block_x = data->args.hipExtLaunchKernel.dimBlocks.x;
+      event.kernel_info.block_y = data->args.hipExtLaunchKernel.dimBlocks.y;
+      event.kernel_info.block_z = data->args.hipExtLaunchKernel.dimBlocks.z;
+      event.kernel_info.grid_x = data->args.hipExtLaunchKernel.numBlocks.x;
+      event.kernel_info.grid_y = data->args.hipExtLaunchKernel.numBlocks.y;
+      event.kernel_info.grid_z = data->args.hipExtLaunchKernel.numBlocks.z;
+      event.kernel_info.func_ptr = (void*)func_addr;
+      event.device_id = hipGetStreamDeviceId(stream);
+
```

Comment:
Could you delete this line? I'd expect our autoformat-on-import to catch this but it doesn't for some reason.

Reply:
Note that this is blocking the merge.

Reply:
the same code is at line 511-513.  i wonder why the issue is not raised over there..
let me remove constantness and see how it comes up...
event.kernel_info.func_ptr = const_cast<void*>(func_addr);

Reply:
event.device_id is vital here to place the kernel at right device, it can't be ignored.

Reply:
Sorry, there was a misunderstanding here. The issue is the extra blank line, not the line with event.device_id

Reply:
deleted ....